app/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDao.java

```diff
@@ -26,156 +26,134 @@
 
-        // It doesn't currently matter which table your consent is in, you can't consent again 
-        // if a record exists.
-        UserConsent existingConsent = getUserConsent(healthCode, new StudyIdentifierImpl(studyConsent.getStudyKey()));
-        if (existingConsent != null) {
+        UserConsent activeConsent = getActiveUserConsent(healthCode, new StudyIdentifierImpl(studyConsent.getStudyKey()));
+        if (activeConsent != null) {
             throw new BridgeServiceException("Consent already exists.", HttpStatus.SC_CONFLICT);
         }
         
         DynamoUserConsent3 consent = new DynamoUserConsent3(healthCode, studyConsent.getStudyKey());
         consent.setConsentCreatedOn(studyConsent.getCreatedOn());
         consent.setSignedOn(signedOn);
-        mapperV3.save(consent);
+        mapper.save(consent);
 
         return consent;
     }
 
     @Override
-    public boolean withdrawConsent(String healthCode, StudyIdentifier studyIdentifier) {
-        // In first step of migration, if user withdraws consent, we delete both consent records. In the future we 
-        // will updated UserConsent3 record with a timestamp, and these will not be returned by queries for an 
-        // active consent.
-        boolean hasWithdrawn = false;
+    public void withdrawConsent(String healthCode, StudyIdentifier studyIdentifier, long withdrewOn) {
+        checkArgument(isNotBlank(healthCode));
+        checkNotNull(studyIdentifier);
+        checkArgument(withdrewOn > 0L);
         
-        DynamoUserConsent2 consent2 = getUserConsent2(healthCode, studyIdentifier.getIdentifier());
-        if (consent2 != null) {
-            mapperV2.delete(consent2);
-            hasWithdrawn = true;
-        }
-        DynamoUserConsent3 consent3 = getUserConsent3(healthCode, studyIdentifier.getIdentifier());
-        if (consent3 != null) {
-            mapperV3.delete(consent3);
-            hasWithdrawn = true;
+        DynamoUserConsent3 activeConsent = (DynamoUserConsent3)getActiveUserConsent(healthCode, studyIdentifier);
+        if (activeConsent == null) {
+            throw new BridgeServiceException("Consent not found.", HttpStatus.SC_NOT_FOUND);
         }
-        return hasWithdrawn;
+        activeConsent.setWithdrewOn(withdrewOn);
+        mapper.save(activeConsent);
     }
 
     @Override
     public boolean hasConsented(String healthCode, StudyIdentifier studyIdentifier) {
-        return getUserConsent(healthCode, studyIdentifier) != null;
+        checkArgument(isNotBlank(healthCode));
+        checkNotNull(studyIdentifier);
+        
+        return getActiveUserConsent(healthCode, studyIdentifier) != null;
     }
 
     @Override
-    public UserConsent getUserConsent(String healthCode, StudyIdentifier studyIdentifier) {
-        UserConsent consent = getUserConsent3(healthCode, studyIdentifier.getIdentifier());
+    public UserConsent getActiveUserConsent(String healthCode, StudyIdentifier studyIdentifier) {
```

Comment:
I see unit tests for (1) it's there, (2) it's there but withdrawn, but not (3) it's not there. Can you add those?